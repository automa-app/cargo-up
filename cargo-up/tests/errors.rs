use insta::assert_snapshot;
mod utils;

#[test]
fn test_package_not_found() {
    let err = utils::run_err("../fixtures/minimum_version/on", &["up", "dep", "notfound"]);
    assert_snapshot!(err);
}

#[test]
fn test_no_upgrader() {
    let err = utils::run_err("../fixtures/minimum_version/on", &["up", "dep", "upgradee"]);
    assert_snapshot!(err);
}

#[test]
fn test_bad_metadata() {
    let err = utils::run_err("../fixtures/bad_metadata/on", &["up", "dep", "clap"]);
    assert_snapshot!(err);
}
