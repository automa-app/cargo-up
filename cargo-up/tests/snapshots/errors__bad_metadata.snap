---
source: tests/errors.rs
expression: err
---
Error during execution of `cargo metadata`:     Updating crates.io index
error: failed to select a version for the requirement `clap = "^0.3.8"`
candidate versions found which didn't match: 2.33.3, 2.33.2, 2.33.1, ...
location searched: crates.io index
required by package `errors v0.1.0 (/Users/pksunkara/Coding/pksunkara/cargo-up/fixtures/bad_metadata/on)`


